### PR TITLE
Sep6 21 add account edit and deletion functionality

### PR DESCRIPTION
**Pull request by KhaledHammoun**

### CODE REVIEW

File: backend/src/main/java/com/sep6/backend/controller/AccountsController.java [hunk 1]
```diff
@@ -1,18 +1,27 @@
...
 @RequestMapping("/accounts")
+@AllArgsConstructor
 public class AccountsController {
 
     private AccountsService service;
-    @Autowired
-    public AccountsController(AccountsService service) {
-        this.service = service;
+
+    @PostMapping(value = "{id}")
```
Comment (tomondre) [inline backend/src/main/java/com/sep6/backend/controller/AccountsController.java:17]: Should we use PUT request in the edit method? The PUT is currently used for other edit methods

File: backend/src/main/java/com/sep6/backend/controller/AccountsController.java [hunk 2]
```diff
@@ -1,18 +1,27 @@
...
+    public void editAccount(@PathVariable int id, @RequestBody Account account){
+        //TODO: return status codes
```
Comment (tomondre) [inline backend/src/main/java/com/sep6/backend/controller/AccountsController.java:19]: We can create exception handling according to https://spring.io/blog/2013/11/01/exception-handling-in-spring-mvc

**Review by tomondre — CHANGES_REQUESTED** [review]

File: backend/src/main/java/com/sep6/backend/controller/AccountsController.java (hunk 2 above)
Comment (tomondre) [reply backend/src/main/java/com/sep6/backend/controller/AccountsController.java:19]: But this should be done in a new task

File: backend/src/main/java/com/sep6/backend/controller/AccountsController.java (hunk 1 above)
Comment (KhaledHammoun) [reply backend/src/main/java/com/sep6/backend/controller/AccountsController.java:17]: Usually put is used when we send just few fields and not the whole object
But I will change if if already all of the rest use put

File: backend/src/main/java/com/sep6/backend/controller/AccountsController.java (hunk 2 above)
Comment (KhaledHammoun) [reply backend/src/main/java/com/sep6/backend/controller/AccountsController.java:19]: Yep there will be separate task for this

**Pull request merged**